lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -158,16 +158,28 @@ def higher_item
           higher_items(1).first
         end
 
+        def higher_items_relation
+          ids = higher_items.map(&:id)
+          acts_as_list_list.where(id: ids).reorder("#{quoted_position_column_with_table_name} DESC")
+        end
+
+        def lower_items_relation
+          ids = lower_items.map(&:id)
+          acts_as_list_list.where(id: ids).reorder("#{quoted_position_column_with_table_name} ASC")
+        end
+
         # Return the next n higher items in the list
         # selects all higher items by default
         def higher_items(limit=nil)
-          limit ||= acts_as_list_list.count
+          limit ||= acts_as_list_list.length
           position_value = send(position_column)
-          acts_as_list_list.
-            where("#{quoted_position_column_with_table_name} <= ?", position_value).
-            where("#{quoted_table_name}.#{self.class.primary_key} != ?", self.send(self.class.primary_key)).
-            reorder("#{quoted_position_column_with_table_name} DESC").
-            limit(limit)
+          acts_as_list_list.select do |item|
+            item.send(position_column).try(:<=, position_value)
+          end.select do |item|
+            item.send(self.class.primary_key).try(:!=, self.send(self.class.primary_key))
+          end.sort do |a, b|
+            b.send(position_column) <=> a.send(position_column)
+          end.first(limit)
```

Comment:
Even on a very small number of records, the memory usage is an order of magnitude more with my PR than without. What do you think about a method that enables eager loading, similar to the way [`Enumerable#lazy`](http://ruby-doc.org/core-2.3.1/Enumerable.html#method-i-lazy) works?